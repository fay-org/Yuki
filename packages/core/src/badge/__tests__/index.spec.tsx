import { mount } from '@vue/test-utils'
import Badge from '..'

describe('Badge', () => {
  it('render normal', () => {
    const wrapper = mount(Badge)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('type', () => {
    const wrapper = mount({
      render() {
        return (
          <div>
            <Badge scale={2 / 3} type="default" />
            <Badge type="secondary" />
            <Badge type="success" />
            <Badge type="warning" />
            <Badge type="error" />
          </div>
        )
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('dot', () => {
    const wrapper = mount(Badge, {
      props: {
        dot: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
