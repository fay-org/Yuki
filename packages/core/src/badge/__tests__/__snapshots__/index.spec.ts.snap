// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should be supprot noramlize props 1`] = `
"<div class="container">
  <div class="fect-dot fect-badge__dot" style="--dot-font-size: calc(1 * 16px); --dot-width: auto; --dot-height: auto; --dot-pt: 0; --dot-pr: 0; --dot-pb: 0; --dot-pl: 0; --dot-mt: 0; --dot-mr: 0; --dot-mb: 0; --dot-ml: 0; --dot-color: #0070f3;" type-data="success"><span class="fect-dot__circle fect-dot__circle--success"></span><span class="fect-dot__ctx"><!----></span></div>
</div>"
`;
